libbeat/cmd/instance/beat.go

```diff
@@ -754,6 +826,8 @@ func (b *Beat) registerTemplateLoading() error {
 				return err
 			}
 			elasticsearch.RegisterConnectCallback(callback)
+		} else if b.Config.ILM.Enabled() {
+			return errors.New("templates cannot be disable when using ILM")
```

Comment:
The difference between the two for me is that the template settings I must overwrite / set to create a valid template. In most cases I do not expect users that they modified the template name / pattern but I still need to overwrite it. If the template is disabled, the user modified it on purpose and I rather abort and tell the user to fix it.

Based on the above it could be argued that I should check if the defaults for modified for the name or pattern. But it can't really made a difference if the default value applies or if it was uncommented.

This also applies to your above comment that I should only error out of ILM is not enabled. I rather have users that made special configs disable these configs first and then start using ILM.